From 005923710c3da36667737acdbdad68b5b3a0c1bc Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 23 Dec 2025 18:28:29 -0800
Subject: [PATCH] build: fix compile simdutf failed on ppc64

...
|subprojects/simdutf/simdutf.cpp:20533:7: error: there are no arguments to 'write_v_u16_11bits_to_utf8' that depend on a template parameter, so a declaration of 'write_v_u16_11bits_to_utf8' must be available [-Wtemplate-body]
|20533 |       write_v_u16_11bits_to_utf8(
|      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~
|subprojects/simdutf/simdutf.cpp:20533:7: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
|subprojects/simdutf/simdutf.cpp:20559:7: error: there are no arguments to 'ppc64_convert_utf16_to_1_2_3_bytes_of_utf8' that depend on a template parameter, so a declaration of 'ppc64_convert_utf16_to_1_2_3_bytes_of_utf8' must be available [-Wtemplate-body]
|20559 |       ppc64_convert_utf16_to_1_2_3_bytes_of_utf8(
|      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|subprojects/simdutf/simdutf.cpp:20533:33: error: 'write_v_u16_11bits_to_utf8' was not declared in this scope
|20533 |       write_v_u16_11bits_to_utf8(
|      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~^
|20534 |           in, utf8_output, as_vector_u8(one_byte_bytemask), one_byte_bitmask);
|      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...

The source file subprojects/simdutf/simdutf.cpp is generated by python3
script singleheader/amalgamate.py and the definition of write_v_u16_11bits_to_utf8
and ppc64_convert_utf16_to_1_2_3_bytes_of_utf8 for ppc64 [1] requires option
--with-utf16 and --with-latin1

Add option --with-utf16 and --with-latin1 by default.

[1] https://github.com/simdutf/simdutf/commit/497f4f29fe25a79af00ce25693a14682a5f669a5

Upstream-Status: Submitted [https://gitlab.gnome.org/GNOME/vte/-/issues/2926]
Tweak for release tarball

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 subprojects/packagefiles/simdutf/meson.build | 3 ++-
 subprojects/simdutf/meson.build              | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/subprojects/packagefiles/simdutf/meson.build b/subprojects/packagefiles/simdutf/meson.build
index 330ee89..ab934c3 100644
--- a/subprojects/packagefiles/simdutf/meson.build
+++ b/subprojects/packagefiles/simdutf/meson.build
@@ -41,11 +41,12 @@ simdutf_amalgamate = [
   '--no-zip',
   '--no-readme',
   '--with-utf8',
-  # '--with-utf16',
+  '--with-utf16',
   '--with-utf32',
   # '--with-base64',
   # '--with-detect-enc',
   '--with-ascii',
+  '--with-latin1',
 ]
 
 if get_option('debug')
diff --git a/subprojects/simdutf/meson.build b/subprojects/simdutf/meson.build
index 330ee89..ab934c3 100644
--- a/subprojects/simdutf/meson.build
+++ b/subprojects/simdutf/meson.build
@@ -41,11 +41,12 @@ simdutf_amalgamate = [
   '--no-zip',
   '--no-readme',
   '--with-utf8',
-  # '--with-utf16',
+  '--with-utf16',
   '--with-utf32',
   # '--with-base64',
   # '--with-detect-enc',
   '--with-ascii',
+  '--with-latin1',
 ]
 
 if get_option('debug')
-- 
2.34.1

